azure-quantum/azure/quantum/optimization/term.py

```diff
@@ -69,10 +71,12 @@ def _convert_if_numpy_type(param: WArray):
         return param
 
 
-class Term:
+class TermBase(ABC):
+    """
+    Term base class; this class is not directly initialized
+    """
     def __init__(
         self,
-        indices: List[int] = None,
         w: Optional[WArray] = None,
```

Comment:
on second glance, this is not being used for serialization so we can remove the need for it in TermBase, but keep it in Term for now.

Reply:
If we want to print that user friendly message for "w" or "c", we can print it in "Term".